wemake_python_styleguide/visitors/ast/classes/classdef.py

```diff
@@ -0,0 +1,161 @@
+                isinstance(sub_node, ast.Name | ast.Attribute | ast.Subscript)
+                for sub_node in parts
+            )
+
+            return len(subscripts) == 1 and correct_items
+        return False
+
+    def _check_base_classes_rules(
+        self,
+        node: ast.ClassDef,
+        base_name: ast.expr,
+    ) -> None:
+        id_attr = getattr(base_name, 'id', None)
+
+        if id_attr == 'BaseException':
+            self.add_violation(bp.BaseExceptionSubclassViolation(node))
+        elif classes.is_forbidden_super_class(
+            id_attr,
+        ) and not enums.has_enum_base(node):
+            self.add_violation(
+                oop.BuiltinSubclassViolation(node, text=id_attr),
+            )
+
+    def _check_kwargs_unpacking(self, node: ast.ClassDef) -> None:
+        for keyword in node.keywords:
+            if keyword.arg is None:
+                self.add_violation(
+                    bp.KwargsUnpackingInClassDefinitionViolation(node),
+                )
+
+
+@final
+class WrongClassBodyVisitor(base.BaseNodeVisitor):
+    """
+    This class is responsible for restricting some ``class`` body anti-patterns.
+
+    Here we check for stylistic issues and design patterns.
+    """
+
+    _allowed_body_nodes: ClassVar[types.AnyNodes] = (
+        *FunctionNodes,
+        ast.ClassDef,  # we allow some nested classes
+        *AssignNodes,  # fields and annotations
+    )
+
+    def visit_ClassDef(self, node: ast.ClassDef) -> None:
+        """Checking class definitions."""
+        self._check_wrong_body_nodes(node)
+        self._check_getters_setters_methods(node)
+        self.generic_visit(node)
+
+    def _check_wrong_body_nodes(self, node: ast.ClassDef) -> None:
+        for sub_node in node.body:
+            if isinstance(sub_node, self._allowed_body_nodes):
+                continue
+            if strings.is_doc_string(sub_node):
+                continue
+            self.add_violation(oop.WrongClassBodyContentViolation(sub_node))
+
+    def _check_getters_setters_methods(self, node: ast.ClassDef) -> None:
+        getters_and_setters = set(
+            getters_setters.find_paired_getters_and_setters(node),
+        ).union(
+            set(  # To delete duplicated violations
+                getters_setters.find_attributed_getters_and_setters(node),
+            ),
+        )
+        for method in getters_and_setters:
+            self.add_violation(
+                oop.UnpythonicGetterSetterViolation(
+                    method,
+                    text=method.name,
+                ),
+            )
+
+
+class ConsecutiveDefaultTypeVarsVisitor(base.BaseNodeVisitor):
+    """Responsible for finding TypeVarTuple after a TypeVar with default."""
+
+    def visit_ClassDef(  # pragma: >=3.13 cover
+        self, node: ast.ClassDef
+    ) -> None:
+        """Check class definition for violation."""
+        if hasattr(node, 'type_params'):  # pragma: no cover
```

Comment:
Please, don't use `no cover` here, we have an exact version, when `type_params` appeared: 3.12+